### PR TITLE
Adding test of different methods of creating objects in JS.

### PR DESCRIPTION
**Pull request by ALEXKIRNAS**
Adding test of different methods of creating objects in JS. 
Compare speed of creating + using objects.
Creating objects using:
- mixing
- classes
- prototypes
- closures (maybe here I make some an error, because sometime closures faster that prototypes)

In my realization time of runs:
prototypes <= classes < closures <= mixing

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
+parserOptions:
```
Comment (aqrln) [inline .eslintrc.yml]: This file is already in the repo. How come you have it added in your commit?

Comment (aqrln) [inline .eslintrc.yml]: And even if there wasn't one, it would be a matter of a different commit and a different PR.

File: JavaScript/c-obj-create.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+const creaters = {};
+module.exports = creaters;
+
+const benchmark = require('./2-benchmark.js');
+global.arraySize = 1000; // Size of arrays in tests
```
Comment (aqrln) [inline JavaScript/c-obj-create.js]: `global`?

File: JavaScript/d-obj-use.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  actions(array);
+}
+
+console.log('Creation + using arrays. Size = ' + arraySize + ':\n');
+benchmark.do(1000, 3, [
+  mixinUse,
+  classUse,
+  prototypeUse,
+  closureUse
+]);
```
Comment (aqrln) [inline JavaScript/d-obj-use.js]: No newline at the end of file.

**Comment by aqrln** [comment]
@ALEXKIRNAS thanks for your contribution! Feel free to ping me if you have any questions or you'll need help with fixing any of the issues and nits mentioned above or the ones that may be found during the further review.

File: .eslintrc.yml (hunk 1 above)
Comment (tshemsedinov) [inline .eslintrc.yml]: It would be great also to run eslint, not only commit it's config

**Comment by ALEXKIRNAS** [comment]
@aqrln I already fixed problems, that you described. And I have a question. I want to test a speed of creating object in file "c-obj-create.js" when I run it like "node c-obj-create.js". But I wan`t to see this test when I use this file as module in other programs. Have can I do this?

**Comment by aqrln** [comment]
@ALEXKIRNAS probably the easiest way to do it is

```js
if (module.parent === null) {
  // this code will only be executed when the file is run as a script,
  // not when required by other modules
}
```

**Comment by aqrln** [comment]
> I already fixed problems, that you described.

Not all of them. The `.eslintrc.yml` file is still present in the PR and I still ask you to rebase your branch on top of `master` :)

**Comment by ALEXKIRNAS** [comment]
@aqrln Fixed. And another question.  How I can add name to functions? Now my version looks like this:
`creaters.mixinCreate = function mixinCreate() {
  // Here my code
};`

**Comment by aqrln** [comment]
@ALEXKIRNAS well, that's the only way. V8 is pretty good at constructing function names itself for stacktraces, but `fn.name` works with named functions only.

**Comment by aqrln** [comment]
I'd prefer the benchmarking framework to not rely upon `fn.name` though.

**Comment by ALEXKIRNAS** [comment]
@aqrln It`s easy to fix up. We can add new argument (array of function`s names) when we call function. Then inside benchmark.do function we choose name for displaying in next order:
1. If to function call added array of names we choose from it.
2. If function called without array of names we trying to get name from fn.name
3. If point 1 and 2 is failed then we naming function some default way, like 'Anonymius'.
I can modify benchmark file in this way.

**Comment by aqrln** [comment]
@ALEXKIRNAS PRs are welcome, go for it :)

**Comment by ALEXKIRNAS** [comment]
@aqrln Done.

**Comment by aqrln** [comment]
@ALEXKIRNAS it must be a different PR.

**Comment by ALEXKIRNAS** [comment]
@aqrln You mean create new branch and make new PR?

**Comment by aqrln** [comment]
Yep. This PR is about object creation benchmarking, not about the benchmarking tools.

**Comment by ALEXKIRNAS** [comment]
@aqrln Okay. I delete it from this PR and add new PR.

File: JavaScript/d-obj-create.js [hunk 4]
```diff
@@ -156,11 +156,16 @@ creaters.closureCreate = function closureCreate() {
...
     creaters.prototypeCreate,
     creaters.closureCreate
+  ], [
+    'mixinCreate',
+    'classCreate',
+    'prototypeCreate',
+    'closureCreate'
   ]);
-}
+}
```
Comment (aqrln) [inline JavaScript/d-obj-create.js]: I think you should configure your editor properly so that it

a) inserts newlines at end of files automatically and doesn't even show them to you, like any source code editor is actually supposed to do, and
b) runs eslint automatically on each save and shows you warnings and errors on the left of lines that have some issues, like any source code editor can do if you tell it to :)

File: JavaScript/d-obj-create.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ module.exports = creaters;
 creaters.arraySize = 1000; // Size of arrays in tests
 
 // Create array that consist of arraySize elements using mixins
-creaters.mixinCreate = function mixinCreate() {
+creaters.mixinCreate = () => {
```
Comment (aqrln) [inline JavaScript/d-obj-create.js]: I feel like this change is appropriate until https://github.com/HowProgrammingWorks/Benchmark/pull/10 lands. This PR is likely to be merged earlier, so let it work with the current benchmarking harness.

**Comment by ALEXKIRNAS** [comment]
Fixed
